app/queues/dns/update-record-flow.server.ts

```diff
@@ -0,0 +1,91 @@
+import { FlowProducer } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { WorkType } from './add-record-flow.server';
+import { updateRecordById } from '~/models/record.server';
+
+import type { Record } from '@prisma/client';
+import type { FlowJob } from 'bullmq';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+import type { SubDomain } from './add-record-flow.server';
+
+export type UpdateDnsRequestData = Pick<Record, 'id' | 'username' | 'type' | 'value'> &
+  Partial<Pick<Record, 'description' | 'course' | 'ports'>> &
```

Comment:
You've become our resident TS master!  I love it.

Reply:
I feel I'm lucky to go through all this!